src/lib/action/entry-transform-to-type.ts

```diff
@@ -0,0 +1,123 @@
+    this.targetContentTypeId = entryTransformation.targetContentType
+    this.identityKey = entryTransformation.identityKey
+    this.shouldPublish = entryTransformation.shouldPublish || false
+    this.removeOldEntries = entryTransformation.removeOldEntries || false
+    this.updateReferences = entryTransformation.updateReferences || false
+    this.transformEntryForLocale = entryTransformation.transformEntryForLocale
+  }
+
+  async applyTo (api: OfflineAPI) {
+    const entries: Entry[] = await api.getEntriesForContentType(this.sourceContentTypeId)
+    const locales: string[] = await api.getLocalesForSpace()
+
+    for (const entry of entries) {
+      const inputs = this.fromFields ? _.pick(entry.fields, this.fromFields) : entry.fields
+      const newEntryId = await this.identityKey(inputs)
+      const hasEntry = await api.hasEntry(newEntryId)
+
+      let skipEntry = true
+      let fieldsForTargetEntry = {}
+
+      for (const locale of locales) {
+        let outputsForCurrentLocale
+        try {
+          outputsForCurrentLocale = await this.transformEntryForLocale(inputs, locale)
+        } catch (err) {
+          await api.recordRuntimeError(err)
+          continue
+        }
+
+        if (outputsForCurrentLocale === undefined) {
+          continue
+        }
+
+        skipEntry = false
+
+        // we collect all the values for the target entry before writing it to the
+        // offline API because we don't yet know if the entry might be skipped
+        // TODO: verify that the derivedFields actually get written to
+        // and to no other field
+        for (const [fieldId, localizedValue] of _.entries(outputsForCurrentLocale)) {
+          if (!fieldsForTargetEntry[fieldId]) {
+            fieldsForTargetEntry[fieldId] = {}
+          }
+          fieldsForTargetEntry[fieldId][locale] = localizedValue
+        }
+      }
+
+      // if derive returned undefined for all locales of this entry, there are no changes
+      // to be made, neither on the source nor the target entry, so we move on to the next
+      if (skipEntry) {
+        continue
+      }
+
+      if (!hasEntry) {
+        const targetEntry = await api.createEntry(this.targetContentTypeId, newEntryId)
+
+        // we are not skipping this source entry and the target entry does not yet exist,
+        // so now is the time to write the collected target entry values to the offline API
+        for (const [fieldId, localizedField] of _.entries(fieldsForTargetEntry)) {
+          if (!targetEntry.fields[fieldId]) {
+            targetEntry.setField(fieldId, {})
+          }
+
+          for (const [locale, localizedValue] of _.entries(localizedField)) {
+            targetEntry.setFieldForLocale(fieldId, locale, localizedValue)
+          }
+
+        }
+        await api.saveEntry(targetEntry.id)
+        if (this.shouldPublish === true || (this.shouldPublish === 'preserve' && entry.isPublished) ) {
```

Comment:
Could you also port this behavior to the standard entry transform action?
Otherwise I fear the APIs diverge too much, plus it's a very useful feature to have.

Reply:
Done